2019-11-27 14:57:22.315  INFO 26284 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 26284 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 14:57:22.317  INFO 26284 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 14:57:23.314  INFO 26284 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 14:57:23.335  INFO 26284 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 14:57:23.335  INFO 26284 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 14:57:23.403  INFO 26284 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 14:57:23.403  INFO 26284 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1042 ms
2019-11-27 14:57:23.443  INFO 26284 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 14:57:23.471  INFO 26284 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 14:57:23.565  WARN 26284 --- [main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'testController': Unsatisfied dependency expressed through field 'webScoketClient'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'scoketClient': Unsatisfied dependency expressed through field 'webSocketClient'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.java_websocket.client.WebSocketClient' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
2019-11-27 14:57:23.576  INFO 26284 --- [main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2019-11-27 14:57:23.588  INFO 26284 --- [main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2019-11-27 14:57:23.699 ERROR 26284 --- [main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Field webSocketClient in cn.kang.socket.ScoketClient required a bean of type 'org.java_websocket.client.WebSocketClient' that could not be found.

The injection point has the following annotations:
	- @org.springframework.beans.factory.annotation.Autowired(required=true)


Action:

Consider defining a bean of type 'org.java_websocket.client.WebSocketClient' in your configuration.

2019-11-27 14:58:19.881  INFO 25428 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 25428 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 14:58:19.884  INFO 25428 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 14:58:20.762  INFO 25428 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 14:58:20.783  INFO 25428 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 14:58:20.783  INFO 25428 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 14:58:20.849  INFO 25428 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 14:58:20.849  INFO 25428 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 932 ms
2019-11-27 14:58:20.888  INFO 25428 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 14:58:20.916  INFO 25428 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 14:58:21.019  WARN 25428 --- [main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'testController': Unsatisfied dependency expressed through field 'webScoketClient'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'scoketClient': Unsatisfied dependency expressed through field 'webSocketClient'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.java_websocket.client.WebSocketClient' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
2019-11-27 14:58:21.031  INFO 25428 --- [main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2019-11-27 14:58:21.044  INFO 25428 --- [main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2019-11-27 14:58:21.120 ERROR 25428 --- [main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

Field webSocketClient in cn.kang.socket.ScoketClient required a bean of type 'org.java_websocket.client.WebSocketClient' that could not be found.

The injection point has the following annotations:
	- @org.springframework.beans.factory.annotation.Autowired(required=true)


Action:

Consider defining a bean of type 'org.java_websocket.client.WebSocketClient' in your configuration.

2019-11-27 15:02:51.444  INFO 23652 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 23652 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:02:51.446  INFO 23652 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:02:52.373  INFO 23652 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:02:52.394  INFO 23652 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:02:52.394  INFO 23652 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:02:52.463  INFO 23652 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:02:52.463  INFO 23652 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 985 ms
2019-11-27 15:02:52.501  INFO 23652 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:02:52.505  INFO 23652 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:02:52.857  INFO 23652 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:02:52.873  INFO 23652 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:02:52.876  INFO 23652 --- [main] cn.kang.Application                      : Started Application in 1.71 seconds (JVM running for 2.341)
2019-11-27 15:03:00.489  INFO 23652 --- [http-nio-8888-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:03:00.491  INFO 23652 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:03:00.498  INFO 23652 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 7 ms
2019-11-27 15:03:26.821  WARN 23652 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:03:26.821  WARN 23652 --- [nioEventLoopGroup-3-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:03:26.821  WARN 23652 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:03:30.131  INFO 25728 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 25728 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:03:30.133  INFO 25728 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:03:31.027  INFO 25728 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:03:31.047  INFO 25728 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:03:31.048  INFO 25728 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:03:31.115  INFO 25728 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:03:31.115  INFO 25728 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 951 ms
2019-11-27 15:03:31.154  INFO 25728 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:03:31.157  INFO 25728 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:03:31.574  INFO 25728 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:03:31.589  INFO 25728 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:03:31.592  INFO 25728 --- [main] cn.kang.Application                      : Started Application in 1.737 seconds (JVM running for 2.349)
2019-11-27 15:03:36.862  INFO 25728 --- [http-nio-8888-exec-2] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:03:36.864  INFO 25728 --- [http-nio-8888-exec-2] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:03:36.870  INFO 25728 --- [http-nio-8888-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initialization in 6 ms
2019-11-27 15:03:37.219  WARN 25728 --- [nioEventLoopGroup-3-1] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

java.lang.ClassCastException: class io.netty.handler.codec.http.HttpObjectAggregator$AggregatedFullHttpRequest cannot be cast to class io.netty.buffer.ByteBuf (io.netty.handler.codec.http.HttpObjectAggregator$AggregatedFullHttpRequest and io.netty.buffer.ByteBuf are in unnamed module of loader 'app')
	at cn.kang.netty.handler.NettyConnectServerHandler.channelRead(NettyConnectServerHandler.java:19) ~[classes/:na]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:306) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:328) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:302) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1422) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:931) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:04:57.490  WARN 25728 --- [nioEventLoopGroup-3-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:04:57.491  WARN 25728 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:04:57.491  WARN 25728 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:05:01.113  INFO 25780 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 25780 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:05:01.116  INFO 25780 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:05:02.086  INFO 25780 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:05:02.108  INFO 25780 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:05:02.109  INFO 25780 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:05:02.177  INFO 25780 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:05:02.177  INFO 25780 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1000 ms
2019-11-27 15:05:02.216  INFO 25780 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:05:02.220  INFO 25780 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:05:02.606  INFO 25780 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:05:02.623  INFO 25780 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:05:02.625  INFO 25780 --- [main] cn.kang.Application                      : Started Application in 1.917 seconds (JVM running for 2.534)
2019-11-27 15:05:03.487  INFO 25780 --- [http-nio-8888-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:05:03.488  INFO 25780 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:05:03.493  INFO 25780 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2019-11-27 15:16:39.459  WARN 25780 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:16:39.459  WARN 25780 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:16:44.842  INFO 8476 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 8476 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:16:44.845  INFO 8476 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:16:46.376  INFO 8476 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:16:46.401  INFO 8476 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:16:46.401  INFO 8476 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:16:46.486  INFO 8476 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:16:46.486  INFO 8476 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1599 ms
2019-11-27 15:16:46.530  INFO 8476 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:16:46.536  INFO 8476 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:16:47.026  INFO 8476 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:16:47.062  INFO 8476 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:16:47.067  INFO 8476 --- [main] cn.kang.Application                      : Started Application in 2.621 seconds (JVM running for 3.617)
2019-11-27 15:16:48.284  WARN 8476 --- [nioEventLoopGroup-3-1] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

java.lang.NullPointerException: null
	at cn.kang.netty.channel.ServerChannelInitializer$1.channelRegistered(ServerChannelInitializer.java:49) ~[classes/:na]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRegistered(CombinedChannelDuplexHandler.java:402) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRegistered(ChannelInboundHandlerAdapter.java:45) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRegistered(CombinedChannelDuplexHandler.java:193) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRegistered(DefaultChannelPipeline.java:1395) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRegistered(DefaultChannelPipeline.java:827) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:506) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:416) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:475) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:163) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:510) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:518) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:16:49.830  INFO 8476 --- [http-nio-8888-exec-3] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:16:49.831  INFO 8476 --- [http-nio-8888-exec-3] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:16:49.837  INFO 8476 --- [http-nio-8888-exec-3] o.s.web.servlet.DispatcherServlet        : Completed initialization in 6 ms
2019-11-27 15:16:50.110  WARN 8476 --- [nioEventLoopGroup-3-2] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

java.lang.NullPointerException: null
	at cn.kang.netty.channel.ServerChannelInitializer$1.channelRegistered(ServerChannelInitializer.java:49) ~[classes/:na]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRegistered(CombinedChannelDuplexHandler.java:402) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRegistered(ChannelInboundHandlerAdapter.java:45) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRegistered(CombinedChannelDuplexHandler.java:193) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRegistered(DefaultChannelPipeline.java:1395) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRegistered(DefaultChannelPipeline.java:827) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:506) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:416) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:475) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:163) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:510) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:518) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:16:55.417  WARN 8476 --- [nioEventLoopGroup-3-3] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

java.lang.NullPointerException: null
	at cn.kang.netty.channel.ServerChannelInitializer$1.channelRegistered(ServerChannelInitializer.java:49) ~[classes/:na]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRegistered(CombinedChannelDuplexHandler.java:402) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRegistered(ChannelInboundHandlerAdapter.java:45) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.CombinedChannelDuplexHandler.channelRegistered(CombinedChannelDuplexHandler.java:193) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRegistered(AbstractChannelHandlerContext.java:140) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRegistered(DefaultChannelPipeline.java:1395) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:161) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRegistered(AbstractChannelHandlerContext.java:147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRegistered(DefaultChannelPipeline.java:827) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:506) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:416) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:475) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:163) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:510) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:518) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:11.198  WARN 8476 --- [nioEventLoopGroup-3-3] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:11.204  WARN 8476 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:11.204  WARN 8476 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:14.973  INFO 15116 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 15116 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:17:14.976  INFO 15116 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:17:16.107  INFO 15116 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:17:16.130  INFO 15116 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:17:16.130  INFO 15116 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:17:16.205  INFO 15116 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:17:16.205  INFO 15116 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1186 ms
2019-11-27 15:17:16.252  INFO 15116 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:17:16.257  INFO 15116 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:17:16.633  INFO 15116 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:17:16.650  INFO 15116 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:17:16.653  INFO 15116 --- [main] cn.kang.Application                      : Started Application in 2.033 seconds (JVM running for 2.727)
2019-11-27 15:17:19.684  INFO 15116 --- [http-nio-8888-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:17:19.685  INFO 15116 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:17:19.690  INFO 15116 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2019-11-27 15:17:49.554  WARN 15116 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:49.554  WARN 15116 --- [nioEventLoopGroup-3-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:49.554  WARN 15116 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:17:53.313  INFO 17088 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 17088 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:17:53.317  INFO 17088 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:17:54.542  INFO 17088 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:17:54.563  INFO 17088 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:17:54.563  INFO 17088 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:17:54.636  INFO 17088 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:17:54.636  INFO 17088 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1276 ms
2019-11-27 15:17:54.678  INFO 17088 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:17:54.682  INFO 17088 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:17:55.051  INFO 17088 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:17:55.068  INFO 17088 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:17:55.071  INFO 17088 --- [main] cn.kang.Application                      : Started Application in 2.104 seconds (JVM running for 2.817)
2019-11-27 15:17:59.319  INFO 17088 --- [http-nio-8888-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:17:59.320  INFO 17088 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:17:59.325  INFO 17088 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 4 ms
2019-11-27 15:31:56.368  WARN 17088 --- [nioEventLoopGroup-3-12] io.netty.channel.DefaultChannelPipeline  : An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.

java.io.IOException: 你的主机中的软件中止了一个已建立的连接。
	at java.base/sun.nio.ch.SocketDispatcher.read0(Native Method) ~[na:na]
	at java.base/sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[na:na]
	at java.base/sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:276) ~[na:na]
	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:233) ~[na:na]
	at java.base/sun.nio.ch.IOUtil.read(IOUtil.java:223) ~[na:na]
	at java.base/sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:358) ~[na:na]
	at io.netty.buffer.PooledByteBuf.setBytes(PooledByteBuf.java:247) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1147) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:347) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:700) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:635) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:552) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:514) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:31:56.384  WARN 17088 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:31:56.384  WARN 17088 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:32:03.888  INFO 22308 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 22308 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:32:03.891  INFO 22308 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:32:04.972  INFO 22308 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:32:04.995  INFO 22308 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:32:04.995  INFO 22308 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:32:05.065  INFO 22308 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:32:05.065  INFO 22308 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1133 ms
2019-11-27 15:32:05.104  INFO 22308 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:32:05.108  INFO 22308 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:32:05.441  INFO 22308 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:32:05.472  INFO 22308 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:32:05.476  INFO 22308 --- [main] cn.kang.Application                      : Started Application in 1.876 seconds (JVM running for 2.54)
2019-11-27 15:32:08.852  INFO 22308 --- [http-nio-8888-exec-2] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:32:08.853  INFO 22308 --- [http-nio-8888-exec-2] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:32:08.857  INFO 22308 --- [http-nio-8888-exec-2] o.s.web.servlet.DispatcherServlet        : Completed initialization in 4 ms
2019-11-27 15:33:35.607  WARN 22308 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:33:35.607  WARN 22308 --- [nioEventLoopGroup-3-9] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:33:35.607  WARN 22308 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:33:39.089  INFO 22372 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 22372 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:33:39.094  INFO 22372 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:33:39.984  INFO 22372 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:33:40.004  INFO 22372 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:33:40.004  INFO 22372 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:33:40.071  INFO 22372 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:33:40.072  INFO 22372 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 947 ms
2019-11-27 15:33:40.111  INFO 22372 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:33:40.116  INFO 22372 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:33:40.476  INFO 22372 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:33:40.493  INFO 22372 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:33:40.497  INFO 22372 --- [main] cn.kang.Application                      : Started Application in 1.694 seconds (JVM running for 2.352)
2019-11-27 15:33:42.384  INFO 22372 --- [http-nio-8888-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:33:42.386  INFO 22372 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:33:42.390  INFO 22372 --- [http-nio-8888-exec-1] o.s.web.servlet.DispatcherServlet        : Completed initialization in 4 ms
2019-11-27 15:34:06.566  WARN 22372 --- [nioEventLoopGroup-3-4] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:34:06.566  WARN 22372 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:34:06.566  WARN 22372 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:34:09.957  INFO 10476 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 10476 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:34:09.959  INFO 10476 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:34:10.910  INFO 10476 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:34:10.931  INFO 10476 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:34:10.931  INFO 10476 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:34:11.001  INFO 10476 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:34:11.002  INFO 10476 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1007 ms
2019-11-27 15:34:11.040  INFO 10476 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:34:11.045  INFO 10476 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:34:11.425  INFO 10476 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:34:11.442  INFO 10476 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:34:11.445  INFO 10476 --- [main] cn.kang.Application                      : Started Application in 1.779 seconds (JVM running for 2.414)
2019-11-27 15:34:12.690  INFO 10476 --- [http-nio-8888-exec-3] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:34:12.691  INFO 10476 --- [http-nio-8888-exec-3] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:34:12.697  INFO 10476 --- [http-nio-8888-exec-3] o.s.web.servlet.DispatcherServlet        : Completed initialization in 6 ms
2019-11-27 15:42:31.635  WARN 10476 --- [nioEventLoopGroup-3-3] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:42:31.635  WARN 10476 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:42:31.635  WARN 10476 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:42:35.419  INFO 10720 --- [main] cn.kang.Application                      : Starting Application on PS20190426HEJH with PID 10720 (D:\workSpace\IDEACode\my_netty\target\classes started by Admin in D:\workSpace\IDEACode\my_netty)
2019-11-27 15:42:35.422  INFO 10720 --- [main] cn.kang.Application                      : No active profile set, falling back to default profiles: default
2019-11-27 15:42:36.299  INFO 10720 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8888 (http)
2019-11-27 15:42:36.321  INFO 10720 --- [main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-11-27 15:42:36.321  INFO 10720 --- [main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-11-27 15:42:36.390  INFO 10720 --- [main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-11-27 15:42:36.390  INFO 10720 --- [main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 937 ms
2019-11-27 15:42:36.429  INFO 10720 --- [main] cn.kang.config.RedisConfig               : JedisPool注入成功！！
2019-11-27 15:42:36.433  INFO 10720 --- [main] cn.kang.config.RedisConfig               : redis地址：114.116.175.209:6379
2019-11-27 15:42:36.797  INFO 10720 --- [main] .s.a.AnnotationAsyncExecutionInterceptor : No task executor bean found for async processing: no bean of type TaskExecutor and no bean named 'taskExecutor' either
2019-11-27 15:42:36.814  INFO 10720 --- [main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8888 (http) with context path ''
2019-11-27 15:42:36.817  INFO 10720 --- [main] cn.kang.Application                      : Started Application in 1.687 seconds (JVM running for 2.349)
2019-11-27 15:42:41.365  INFO 10720 --- [http-nio-8888-exec-4] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring DispatcherServlet 'dispatcherServlet'
2019-11-27 15:42:41.366  INFO 10720 --- [http-nio-8888-exec-4] o.s.web.servlet.DispatcherServlet        : Initializing Servlet 'dispatcherServlet'
2019-11-27 15:42:41.371  INFO 10720 --- [http-nio-8888-exec-4] o.s.web.servlet.DispatcherServlet        : Completed initialization in 5 ms
2019-11-27 15:45:09.661  WARN 10720 --- [nioEventLoopGroup-2-1] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

2019-11-27 15:45:09.661  WARN 10720 --- [nioEventLoopGroup-2-2] io.netty.channel.AbstractChannel         : Can't invoke task later as EventLoop rejected it

java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.invokeLater(AbstractChannel.java:1019) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.deregister(AbstractChannel.java:810) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.fireChannelInactiveAndDeregister(AbstractChannel.java:770) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:753) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:608) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.closeAll(NioEventLoop.java:748) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:527) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1044) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-all-4.1.42.Final.jar:4.1.42.Final]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

